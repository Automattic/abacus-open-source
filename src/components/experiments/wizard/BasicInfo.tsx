import { InputAdornment, TextField as MuiTextField, Typography } from '@material-ui/core'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import { Alert, AutocompleteRenderInputParams } from '@material-ui/lab'
import * as dateFns from 'date-fns'
import { Field, useField } from 'formik'
import { TextField } from 'formik-material-ui'
import React from 'react'

import AbacusAutocomplete, { autocompleteInputProps } from 'src/components/general/Autocomplete'
import {
  MAX_DISTANCE_BETWEEN_NOW_AND_START_DATE_IN_MONTHS,
  MAX_DISTANCE_BETWEEN_START_AND_END_DATE_IN_MONTHS,
} from 'src/lib/schemas'
import { formatIsoDate } from 'src/utils/time'

import { ExperimentFormCompletionBag } from './ExperimentForm'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {},
    row: {
      margin: theme.spacing(5, 0, 1, 0),
      display: 'flex',
      alignItems: 'center',
      '&:first-of-type': {
        marginTop: theme.spacing(3),
      },
      [theme.breakpoints.down('xs')]: {
        flexDirection: 'column',
        alignItems: 'stretch',
      },
    },
    through: {
      flex: 0,
      margin: theme.spacing(0, 2),
      color: theme.palette.text.hint,
      [theme.breakpoints.down('xs')]: {
        margin: theme.spacing(2, 2),
      },
    },
    datePicker: {
      flex: 1,
      '& input:invalid': {
        // Fix the native date-picker placeholder text colour
        color: theme.palette.text.hint,
      },
    },
  }),
)

const BasicInfo = ({
  completionBag: { userCompletionDataSource },
}: {
  completionBag: ExperimentFormCompletionBag
}): JSX.Element => {
  const classes = useStyles()

  const [startDateField] = useField('experiment.startDatetime')
  const minStartDate = new Date()
  const maxStartDate = dateFns.addMonths(new Date(), MAX_DISTANCE_BETWEEN_NOW_AND_START_DATE_IN_MONTHS)
  const minEndDate = startDateField.value && new Date(startDateField.value)
  const maxEndDate =
    startDateField.value &&
    dateFns.addMonths(new Date(startDateField.value), MAX_DISTANCE_BETWEEN_START_AND_END_DATE_IN_MONTHS)
  const formatDateForInput = (date: Date) => (date ? formatIsoDate(date) : undefined)

  return (
    <div className={classes.root}>
      <Typography variant='h4' gutterBottom>
        Basic Info
      </Typography>

      <div className={classes.row}>
        <Field
          component={TextField}
          name='experiment.name'
          id='experiment.name'
          label='Experiment name'
          placeholder='experiment_name'
          helperText='Use snake_case.'
          variant='outlined'
          fullWidth
          required
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>

      <div className={classes.row}>
        <Field
          component={TextField}
          name='experiment.description'
          id='experiment.description'
          label='Experiment description'
          placeholder='Monthly vs. yearly pricing'
          helperText='State your hypothesis.'
          variant='outlined'
          fullWidth
          required
          multiline
          rows={4}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>

      <div className={classes.row}>
        <Field
          component={TextField}
          className={classes.datePicker}
          name='experiment.startDatetime'
          id='experiment.startDatetime'
          label='Start date (UTC)'
          type='date'
          variant='outlined'
          required
          InputLabelProps={{
            shrink: true,
          }}
          inputProps={{
            min: formatDateForInput(minStartDate),
            max: formatDateForInput(maxStartDate),
          }}
        />
        <span className={classes.through}> through </span>
        <Field
          component={TextField}
          className={classes.datePicker}
          name='experiment.endDatetime'
          id='experiment.endDatetime'
          label='End date (UTC)'
          type='date'
          variant='outlined'
          required
          InputLabelProps={{
            shrink: true,
          }}
          inputProps={{
            min: formatDateForInput(minEndDate),
            max: formatDateForInput(maxEndDate),
          }}
        />
      </div>
      <Alert severity='info'>
        The experiment will <strong>start and end automatically</strong> around 00:10 UTC on these dates.
      </Alert>

      <div className={classes.row}>
        <Field
          component={AbacusAutocomplete}
          name='experiment.ownerLogin'
          id='experiment.ownerLogin'
          fullWidth
          options={userCompletionDataSource.data ?? []}
          loading={userCompletionDataSource.isLoading}
          noOptionsText='No users found'
          renderInput={(params: AutocompleteRenderInputParams) => (
            <MuiTextField
              {...params}
              label='Owner'
              placeholder='wp_username'
              helperText='Use WordPress.com username.'
              variant='outlined'
              required
              InputProps={{
                ...autocompleteInputProps(params, userCompletionDataSource.isLoading),
                startAdornment: <InputAdornment position='start'>@</InputAdornment>,
              }}
              InputLabelProps={{
                shrink: true,
              }}
            />
          )}
        />
      </div>
    </div>
  )
}

export default BasicInfo
