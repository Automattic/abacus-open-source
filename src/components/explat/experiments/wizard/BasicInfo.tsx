import { InputAdornment, TextField as MuiTextField, Typography } from '@material-ui/core'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import { Alert, AutocompleteRenderInputParams } from '@material-ui/lab'
import { Field } from 'formik'
import { TextField } from 'formik-material-ui'
import React from 'react'

import AbacusAutocomplete, { autocompleteInputProps } from 'src/components/general/Autocomplete'
import PrivateLink from 'src/components/general/PrivateLink'

import { ExperimentFormCompletionBag } from './ExperimentForm'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {},
    row: {
      margin: theme.spacing(5, 0, 1, 0),
      display: 'flex',
      alignItems: 'center',
      '&:first-of-type': {
        marginTop: theme.spacing(3),
      },
      [theme.breakpoints.down('xs')]: {
        flexDirection: 'column',
        alignItems: 'stretch',
      },
    },
    through: {
      flex: 0,
      margin: theme.spacing(0, 2),
      color: theme.palette.text.hint,
      [theme.breakpoints.down('xs')]: {
        margin: theme.spacing(2, 2),
      },
    },
    datePicker: {
      flex: 1,
      '& input:invalid': {
        // Fix the native date-picker placeholder text colour
        color: theme.palette.text.hint,
      },
    },
  }),
)

const BasicInfo = ({
  completionBag: { userCompletionDataSource },
}: {
  completionBag: ExperimentFormCompletionBag
}): JSX.Element => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <Typography variant='h4' gutterBottom>
        Start designing your experiment
      </Typography>

      <Typography variant='body2'>
        Our{' '}
        <PrivateLink underline='always' href='https://wp.me/PCYsg-Hs4' target='_blank'>
          FieldGuide
        </PrivateLink>{' '}
        is a great place to start, it will instruct you on how to{' '}
        <PrivateLink underline='always' href='https://wp.me/PCYsg-FqB' target='_blank'>
          design your experiment.
        </PrivateLink>
        <br />
        <br />
      </Typography>

      <div className={classes.row}>
        <Field
          component={TextField}
          name='experiment.name'
          id='experiment.name'
          label='Experiment name'
          placeholder='experiment_name'
          helperText='Use snake_case.'
          variant='outlined'
          fullWidth
          required
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>

      <div className={classes.row}>
        <Field
          component={TextField}
          name='experiment.description'
          id='experiment.description'
          label='Experiment description'
          placeholder='Monthly vs. yearly pricing'
          helperText='State your hypothesis.'
          variant='outlined'
          fullWidth
          required
          multiline
          rows={4}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>

      <Alert severity='warning'>
        Experiments are no longer schedulable, and must be started/stopped manually via the buttons on the experiment
        page.
      </Alert>

      <div className={classes.row}>
        <Field
          component={AbacusAutocomplete}
          name='experiment.ownerLogin'
          id='experiment.ownerLogin'
          fullWidth
          options={userCompletionDataSource.data ?? []}
          loading={userCompletionDataSource.isLoading}
          noOptionsText='No users found'
          renderInput={(params: AutocompleteRenderInputParams) => (
            <MuiTextField
              {...params}
              label='Owner'
              placeholder='wp_username'
              helperText='Use WordPress.com username.'
              variant='outlined'
              required
              InputProps={{
                ...autocompleteInputProps(params, userCompletionDataSource.isLoading),
                startAdornment: <InputAdornment position='start'>@</InputAdornment>,
              }}
              InputLabelProps={{
                shrink: true,
              }}
            />
          )}
        />
      </div>

      <div className={classes.row}>
        <Field
          component={TextField}
          id='experiment.p2Url'
          name='experiment.p2Url'
          placeholder='https://a8cexperiments.wordpress.com/your-experiment-url'
          label='Your a8cexperiments P2 post URL'
          helperText='Optional for now, but a URL is required for experiment launch.'
          variant='outlined'
          fullWidth
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>
      <Alert severity='info'>
        After submitting your experiment, Abacus will provide a pre-filled P2 experiment template in the Setup tab to
        get you started on{' '}
        <PrivateLink underline='always' href='https://wp.me/PCYsg-Gek' target='_blank'>
          experiment documentation
        </PrivateLink>
        .
      </Alert>
    </div>
  )
}

export default BasicInfo
