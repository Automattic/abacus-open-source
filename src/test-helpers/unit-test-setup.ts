import * as notistack from 'notistack'

/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */
// @ts-ignore
window.URL.createObjectURL = jest.fn()

window.HTMLCanvasElement.prototype.getContext = jest.fn()

jest.mock('notistack')
const mockedNotistack = notistack as jest.Mocked<typeof notistack>
beforeEach(() => {
  mockedNotistack.useSnackbar.mockImplementation(() => ({
    enqueueSnackbar: jest.fn(),
    closeSnackbar: jest.fn(),
  }))
})

global.console = {
  ...console,
  error: jest.fn(),
}
