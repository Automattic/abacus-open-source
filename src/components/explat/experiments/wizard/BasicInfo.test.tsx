import { act, fireEvent, getByLabelText, render } from '@testing-library/react'
import MockDate from 'mockdate'
import React from 'react'

import { MockFormik } from 'src/test-helpers/test-utils'

import BasicInfo from './BasicInfo'
import { ExperimentFormCompletionBag } from './ExperimentForm'

MockDate.set('2020-07-21')

const completionBag: ExperimentFormCompletionBag = {
  userCompletionDataSource: {
    data: null,
    error: null,
    isLoading: false,
    reloadRef: { current: () => undefined },
  },
  eventCompletionDataSource: {
    data: null,
    error: null,
    isLoading: false,
    reloadRef: { current: () => undefined },
  },
  exclusionGroupCompletionDataSource: {
    data: null,
    error: null,
    isLoading: false,
    reloadRef: { current: () => undefined },
  },
}

test('renders as expected', () => {
  const { container } = render(
    <MockFormik>
      <BasicInfo completionBag={completionBag} />
    </MockFormik>,
  )
  expect(container).toMatchSnapshot()
})

test('renders sensible dates as expected', () => {
  MockDate.set('2020-07-21')
  const { container } = render(
    <MockFormik initialValues={{ experiment: { startDatetime: '', endDatetime: '' } }}>
      <BasicInfo completionBag={completionBag} />
    </MockFormik>,
  )
  const startDateInput = getByLabelText(container, /Start date/)
  const endDateInput = getByLabelText(container, /End date/)

  act(() => {
    fireEvent.change(startDateInput, { target: { value: '2020-07-28' } })
    fireEvent.change(endDateInput, { target: { value: '2020-10-28' } })
  })
  expect(container).toMatchSnapshot()
})

test('renders date validation errors as expected', () => {
  MockDate.set('2020-07-21')
  const { container } = render(
    <MockFormik initialValues={{ experiment: { startDatetime: '', endDatetime: '' } }}>
      <BasicInfo completionBag={completionBag} />
    </MockFormik>,
  )
  const startDateInput = getByLabelText(container, /Start date/)
  const endDateInput = getByLabelText(container, /End date/)

  // Start date before today
  fireEvent.change(startDateInput, { target: { value: '2020-07-20' } })
  expect(container).toMatchSnapshot()

  // Start date too far into the future
  fireEvent.change(startDateInput, { target: { value: '2025-07-20' } })
  expect(container).toMatchSnapshot()

  // End date before start date
  act(() => {
    fireEvent.change(startDateInput, { target: { value: '2020-07-28' } })
    fireEvent.change(endDateInput, { target: { value: '2020-07-20' } })
  })
  expect(container).toMatchSnapshot()

  // End date too far into the future
  fireEvent.change(endDateInput, { target: { value: '2025-07-21' } })
  expect(container).toMatchSnapshot()
})
