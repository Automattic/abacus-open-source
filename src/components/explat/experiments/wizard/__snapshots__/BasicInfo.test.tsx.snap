// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      Start designing your experiment
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      Our
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiTypography-colorPrimary"
        target="_blank"
      >
        FieldGuide
      </a>
       
      is a great place to start, it will instruct you on how to
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiTypography-colorPrimary"
        target="_blank"
      >
        design your experiment.
      </a>
      <br />
      <br />
    </p>
    <div
      class="makeStyles-row-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="true"
          for="experiment.name"
          id="experiment.name-label"
        >
          Experiment name
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-describedby="experiment.name-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="experiment.name"
            name="experiment.name"
            placeholder="experiment_name"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Experiment name
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
          id="experiment.name-helper-text"
        >
          Use snake_case.
        </p>
      </div>
    </div>
    <div
      class="makeStyles-row-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="true"
          for="experiment.description"
          id="experiment.description-label"
        >
          Experiment description
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-describedby="experiment.description-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            id="experiment.description"
            name="experiment.description"
            placeholder="Monthly vs. yearly pricing"
            required=""
            rows="4"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Experiment description
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
          id="experiment.description-helper-text"
        >
          State your hypothesis.
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardWarning MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        Experiments are no longer schedulable, and must be started/stopped manually via the buttons on the experiment page.
      </div>
    </div>
    <div
      class="makeStyles-row-2"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        name="experiment.ownerLogin"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="true"
            for="experiment.ownerLogin"
            id="experiment.ownerLogin-label"
          >
            Owner
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                @
              </p>
            </div>
            <input
              aria-autocomplete="list"
              aria-describedby="experiment.ownerLogin-helper-text"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="experiment.ownerLogin"
              placeholder="wp_username"
              required=""
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
              >
                <span>
                  Owner
                   *
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
            id="experiment.ownerLogin-helper-text"
          >
            Use WordPress.com username.
          </p>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-row-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
          data-shrink="true"
          for="experiment.p2Url"
          id="experiment.p2Url-label"
        >
          Your a8cexperiments P2 post URL
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-describedby="experiment.p2Url-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="experiment.p2Url"
            name="experiment.p2Url"
            placeholder="https://a8cexperiments.wordpress.com/your-experiment-url"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Your a8cexperiments P2 post URL
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
          id="experiment.p2Url-helper-text"
        >
          Optional for now, but a URL is required for experiment launch.
        </p>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        After submitting your experiment, Abacus will provide a pre-filled P2 experiment template in the Setup tab to get you started on
         
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiTypography-colorPrimary"
          target="_blank"
        >
          experiment documentation
        </a>
        .
      </div>
    </div>
  </div>
</div>
`;
